proposals/route-blinding.md

```diff
@@ -0,0 +1,285 @@
+# Route Blinding
+
+## Table of Contents
+
+* [Proposal](#proposal)
+  * [Introduction](#introduction)
+  * [Overview](#overview)
+  * [Notations](#notations)
+  * [Requirements](#requirements)
+  * [Encrypted recipient data](#encrypted-recipient-data)
+  * [Creating a blinded route](#creating-a-blinded-route)
+  * [Sending to a blinded route](#sending-to-a-blinded-route)
+  * [Receiving from a blinded route](#receiving-from-a-blinded-route)
+  * [Sample flow](#sample-flow)
+  * [Unblinding channels via fee probing](#unblinding-channels-via-fee-probing)
+* [Tips and Tricks](#tips-and-tricks)
+  * [Recipient pays fees](#recipient-pays-fees)
+  * [Dummy hops](#dummy-hops)
+  * [Wallets and unannounced channels](#wallets-and-unannounced-channels)
+  * [Blinded trampoline route](#blinded-trampoline-route)
+* [FAQ](#faq)
+  * [Why not use rendezvous](#why-not-use-rendezvous)
+  * [Why not use HORNET](#why-not-use-hornet)
+
+## Proposal
+
+### Introduction
+
+Route blinding is a lightweight technique to provide recipient anonymity by blinding an arbitrary
+amount of hops at the end of an onion path. It's more flexible than rendezvous because it lets
+senders arbitrarily update amounts and lock times, and reuse a blinded route multiple times (which
+is useful when retrying a failed route or using multi-part payments).
+
+The downside compared to rendezvous is that senders have more leeway to probe by changing various
+variables, so the scheme needs to explicitly defend against probing attacks and may be less private.
+
+Some use-cases where route blinding is useful include:
+
+* Recipient anonymity when receiving payments
+* Using unannounced channels in invoices without revealing them
+* Forcing a payment to go through a specific set of intermediaries that can witness the payment
+* Providing anonymous reply paths for onion messages (offers, stuckless payments, etc)
+
+### Overview
+
+At a high level, route blinding works by having the recipient choose an introduction point and a
+route to himself from that introduction point. The recipient then blinds each node and channel
+along that route with ECDH. The recipient includes the blinded route and a secret in the invoice,
```

Comment:
```suggestion
along that route with ECDH. The recipient includes the blinded route and a _hop-blinding secret_ in the invoice,
```

This secret needs its own name, otherwise it can get confused with the `payment_secret` which is also referenced further down. We should also make sure to use the unique names whenever possible to avoid confusion.

Reply:
Done in https://github.com/lightningnetwork/lightning-rfc/pull/765/commits/0468ed01e003e1e1ed7e49c0802b5f4c10519cba